codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/rustlang/CargoDependency.kt

```diff
@@ -52,7 +52,7 @@ sealed class RustDependency(open val name: String) : SymbolDependencyContainer {
  *
  * CodegenVisitor deduplicates inline dependencies by (module, name) during code generation.
  */
-class InlineDependency(
+open class InlineDependency(
```

Comment:
These are the main changes in the client codegen. They promote some code re-use while avoiding the client codegen to depend on the server codegen.

Reply:
Why do you want to inherit from this as opposed to creating `ServerInlineDependency` as an `object` since it only has the one companion function on it? Is there a future use-case for polymorphism?

Reply:
Probably this is just my inexperience with Kotlin..I guess what you are saying is that i can just make `forRustFile` public and use it in my `ServerInlineDependency` companion object, avoiding to  open the class..

Thanks for pointing  this out, writing good kotlin is still a challenge for me..

Reply:
I'll fix this ASAP!

Reply:
You can simplify the companion object out entirely:
```kotlin
object ServerInlineDependency {
    fun serverOperationHandler(runtimeConfig: RuntimeConfig): InlineDependency = // ...
}
```

Reply:
Done, looks nice now!